bolero-generator/src/array.rs

```diff
@@ -1,122 +1,54 @@
 use crate::{Driver, TypeGenerator, TypeGeneratorWithParams, TypeValueGenerator, ValueGenerator};
 
-impl<T> TypeGenerator for [T; 0] {
-    fn generate<D_: Driver>(_driver: &mut D_) -> Option<Self> {
-        Some([])
-    }
-}
+impl<T: TypeGenerator, const LEN: usize> TypeGenerator for [T; LEN] {
+    fn generate<D: Driver>(driver: &mut D) -> Option<Self> {
+        // TODO use core::array::try_from_fn once stable https://github.com/rust-lang/rust/issues/89379
+        let mut maybe_init: [Option<T>; LEN] = [(); LEN].map(|_| None);
```

Comment:
Question: Can't we just initialize to `[None; LEN]`? Or why is this needed?

Reply:
No, that array literal syntax requires that the values implement `Clone`, which I don't want to force